File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    answer = 0;
    for i in range(n+1,1000000) :
        
        count1 = bin(n)[2:].count('1');
        count2 = bin(i)[2:].count('1');
        if count1 == count2 :
            return i;
            
        
    

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.js
function solution(n) {
    var answer = 0;
    
    let count1 = 0

    nx = n.toString(2)
    for(let j=0;j<nx.length;j++){
        if(nx[j] == 1) {
            count1 +=1
        }
    }
    
    for(let i=n+1; i < 1000000; i++){
        let count2 = 0
        bin = i.toString(2)

        for(let j=0;j<bin.length;j++){
            if(bin[j] == 1){
                count2 +=1
            }
        }
        if(count1 == count2){
            answer = i
            break
        }
    }
    return answer;
}